DO NOT MERGE Show menubar only when the Alt key is hit

wodim opened the pull request:
This is a tentative implementation for #1915. There's no option at all (it's always on), I just want to know if it works well for others and on OSes other than Linux.

wodim committed:
Show menubar only when the Alt key is hit

un1versal commented:
I dont mind the idea of hiding/unhiding menu bar by pressing Alt, but as is hidden by default is a -1 from me.

FYI works only to unhide it pressing Alt, if you press Alt again to hide it it does sod all. Clicking the main window bar hides it.

un1versal commented on src/kvirc/ui/KviMainWindow.cpp:
```diff
@@ -133,6 +133,7 @@ KviMainWindow::KviMainWindow()
     m_pAccellerators = new KviPointerList<QShortcut>;
 	m_pMenuBar   = new KviMenuBar(this,"main_menu_bar");
 	setMenuWidget(m_pMenuBar);
+	m_pMenuBar->hide();
```
Should not be hidden by default imo,

wodim replied:
Read the pull request.

un1versal replied:
I have, and if this is proposed as is I disagree + also doesnt work well to hide if you already un-hidden it, its counter intuitive.

in #1915 you make sane comments and then propose the exact opposite. Sorry I dont get it.

ctrlaltca replied:
> I have, and if this is proposed as is I disagree
> 
> This is a tentative implementation ... I just want to know if it works well for others and on OSes other than Linux.

Once @wodim's knows this works fine, he'll probably move on trying to implement the missing bits and shaping up this PR better. If this doesn't work on other OSes, he won't waste his time polishing up a not-working thing.

un1versal commented on src/kvirc/ui/KviMainWindow.cpp:
```diff
@@ -917,6 +918,17 @@ void KviMainWindow::contextMenuEvent(QContextMenuEvent *)
+	switch(event->key())
+	{
+		case Qt::Key_Alt:
+			g_pMainWindow->showMenubar();
```
should probably also include

```
else
case Qt::Key_Alt:
            g_pMainWindow->hideMenubar();
```

So you can also hide by pressing Alt, as is you need to click to hide

wodim committed:
Close the menubar with Alt also.

wodim committed:
Add this feature as an option.

wodim committed:
Remove this option on OS X.

For the sake of simplicity I won't comment everything out, the
option will always reset to false on launch and that's it.

un1versal commented:
I suppose you read https://github.com/kvirc/KVIrc/issues/1917 which is the reason I suggested a statusbar indicator. not because of bikeshedding.

wodim commented:
I am thinking of adding an entry to the ircview popup called "show menubar", what do you think? Something that sets `AutoHideMenubar` to false.

un1versal commented:
that maybe useful but not as obvious perhaps, what about a toolbar icon (visible by default), similar to away/notaway that hides/unhides that would probably be more obvious?

I can make the icons 16x16 and 32x32 (4 icons total) if you think thats a good idea and send you a PR to this branch

wodim commented:
I don't want such a button by default on a toolbar.

un1versal commented:
What about it becomes visible in toolbar if you select option OR hide menubar?

that would be 2 icons only in that case

we already have something similar, the tray icon that becomes visible in tray if you minimize kvirc even if the option is unchecked

wodim commented:
Does not make much sense to me since toolbars are customisable by users...

un1versal commented:
well we need some **obvious** indicator that toolbar is hidden, however it makes more sense to do that.

Else support burden is a pita

IF someone complains (wheres my menu gone?), we ask for a screenshot, if the indicator is there we know its because of this option. If indicator not there we know its some other bug related to #1917 
it makes sense to me.

Without this PR, there are 2 causes (1 known 1 unknown) for menu disappearing with this PR there are 3 causes.  (2 known 1 unknown)

wodim committed:
Fix compilation on MSVC
